stock_brand/models/stock_picking.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+#  Copyright (C) 2019 Sunflower IT <sunflowerweb.nl>
+#  License GNU Affero General Public License <http://www.gnu.org/licenses/>.
+
+from odoo import api, fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    brand_id = fields.Many2one(
+        'res.brand', string='Brand',
+        help="Brand to use for this sale")
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    def _action_confirm(self):
+        ret = super(StockMove, self)._action_confirm()
+        for this in self:
+            if this.sale_line_id.order_id:
+                this.picking_id.brand_id = this.sale_line_id.order_id.brand_id
```

Comment:
the brand should be set when the picking is created from the sale order.

I see two issues in this override:

- if the picking contains moves from different sale orders? 
- if the user already set the brand manually, it will erase it.

Please consider use `_action_launch_stock_rule` or simply override the `action_confirm` of the sale order